ovirt/resource_ovirt_vm.go

```diff
@@ -566,6 +579,41 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	isAutoPinning := false
+	// Mimic the UI behavior. unless specified set to existing policy for high performance VMs.
+	if _, ok := d.GetOk("auto_pinning_policy"); !ok && isHighPerformance {
+		err := d.Set("auto_pinning_policy", ovirtsdk4.AUTOPINNINGPOLICY_EXISTING)
+		if err != nil {
+			return err
+		}
+	}
+	if v, ok := d.GetOk("auto_pinning_policy"); ok {
+		isAutoPinning = true
+		autoPinningPolicy := ovirtsdk4.AutoPinningPolicy(fmt.Sprint(v))
+
+		// if we have a policy, we need to set the pinning to all the hosts in the cluster.
+		if autoPinningPolicy != ovirtsdk4.AUTOPINNINGPOLICY_DISABLED {
+			clusterService := conn.SystemService().ClustersService().ClusterService(cluster.MustId())
+			clusterGet, err := clusterService.Get().Send()
+			if err != nil {
+				return err
+			}
+			clusterName := clusterGet.MustCluster().MustName()
+			hostsInCluster, err := conn.SystemService().HostsService().List().Search(
+				fmt.Sprintf("cluster=%s", clusterName)).Send()
+			if err != nil {
+				return err
+			}
+			placementPolicyBuilder := ovirtsdk4.NewVmPlacementPolicyBuilder()
+			placementPolicy, err := placementPolicyBuilder.Hosts(hostsInCluster.MustHosts()).
+				Affinity(ovirtsdk4.VMAFFINITY_MIGRATABLE).Build()
+			if err != nil {
+				return err
```

Comment:
Please add a meaningful error message wrapping the original unless the Build function already gives you a meaningful error message with context.